src/toil/batchSystems/kubernetes.py

```diff
@@ -52,8 +53,25 @@
 from toil.lib.humanize import human2bytes
 from toil.resource import Resource
 
+from toil.lib.retry import retry
+
 logger = logging.getLogger(__name__)
+     
+def retryable_kubernetes_errors(e):
+    """
+    A function that determins whether or not Toil should retry or stop given 
+    exceptions thrown by Kubernetes. 
+    """
+    if isinstance(e, urllib3.exceptions.MaxRetryError) or isinstance(e, urllib3.exceptions.HTTPError(504)) or isinstance(e, ApiException):
+        return False
+    return True
```

Comment:
OK, now I think there are 2 new problems.

1. The sense is backward here. the function is supposed to return True if the error observed is likely to be temporary, and False if it's a permanent error that can't be fixed by retrying. As written, you're retrying for all errors *except* those on the list, so e.g. `NameError` (caused by using an undefined variable) will result in a retry, while `urllib3.exceptions.MaxRetryError` will not result in a retry. I think you need to swap False and True.

2. `urllib3.exceptions.HTTPError(504)` is an instance of `HTTPError` with `504` as its first argument. It is not a class that other exceptions are instances of. You can match all `HTTPError` instances with `isinstance(e, urllib3.exceptions.HTTPError)`. There's no documentation I can find about `urllib3` putting HTTP status codes into `HTTPError` anyway, so I'm not sure it makes any sense to try to catch only the `HTTPError`s that correspond to 504s coming from the server.

How did you decide to catch HTTPError here anyway? It didn't come up in https://github.com/DataBiosphere/toil/issues/2884